app/bundles/DynamicContentBundle/Form/Type/DynamicContentType.php

```diff
@@ -181,22 +256,91 @@ public function buildForm(FormBuilderInterface $builder, array $options)
+            $builder->create(
+                'filters',
+                'collection',
+                [
+                    'type'    => DwcEntryFiltersType::class,
+                    'options' => [
+                        'countries'    => $this->countryChoices,
+                        'regions'      => $this->regionChoices,
+                        'timezones'    => $this->timezoneChoices,
+                        'locales'      => $this->localeChoices,
+                        'fields'       => $this->fieldChoices,
+                        'deviceTypes'  => $this->deviceTypesChoices,
+                        'deviceBrands' => $this->deviceBrandsChoices,
+                        'deviceOs'     => $this->deviceOsChoices,
+                        'tags'         => $this->tagChoices,
+                    ],
+                    'error_bubbling' => false,
+                    'mapped'         => true,
+                    'allow_add'      => true,
+                    'allow_delete'   => true,
+                ]
+            )->addModelTransformer($filterModalTransformer)
+        );
+
         if (!empty($options['action'])) {
             $builder->setAction($options['action']);
         }
+
+        $builder->addEventListener(
+            FormEvents::PRE_SUBMIT,
+            function (FormEvent $event) {
+                // delete default prototype values
+                $data = $event->getData();
+                unset($data['filters']['__name__']);
+                $event->setData($data);
+            }
+        );
     }
 
     /**
      * @param OptionsResolver $resolver
+     *
+     * @throws \Symfony\Component\OptionsResolver\Exception\AccessException
      */
     public function configureOptions(OptionsResolver $resolver)
     {
-        $resolver->setDefaults(['data_class' => 'Mautic\DynamicContentBundle\Entity\DynamicContent']);
+        $resolver->setDefaults([
+            'data_class'     => DynamicContent::class,
+            'label'          => false,
+            'error_bubbling' => false,
+        ]);
 
         $resolver->setDefined(['update_select']);
     }
 
     /**
+     * {@inheritdoc}
+     */
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        $view->vars['fields']       = $this->fieldChoices;
+        $view->vars['countries']    = $this->countryChoices;
+        $view->vars['regions']      = $this->regionChoices;
+        $view->vars['timezones']    = $this->timezoneChoices;
+        $view->vars['deviceTypes']  = $this->deviceTypesChoices;
+        $view->vars['deviceBrands'] = $this->deviceBrandsChoices;
+        $view->vars['deviceOs']     = $this->deviceOsChoices;
+        $view->vars['tags']         = $this->tagChoices;
+        $view->vars['locales']      = $this->localeChoices;
+    }
+
+    private function filterFieldChoices()
+    {
+        $this->fieldChoices['lead'] = array_filter($this->fieldChoices['lead'], function ($key) {
```

Comment:
This should be an inclusive list to return only the filters supported rather than filtering out those we don't because otherwise filters injected by plugins will still show or every time we add a new filter, we'll have to add it here.

Reply:
Also, let's exclude company fields. This feature is for visitors and we won't know their company at that point.